packages/frontend/amp/components/elements/YoutubeBlockComponent.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { textSans } from '@guardian/pasteup/typography';
+import { palette } from '@guardian/pasteup/palette';
+import { css } from 'emotion';
+import { pillarPalette } from '@frontend/lib/pillars';
+import TriangleIcon from '@guardian/pasteup/icons/triangle.svg';
+
+const figureStyle = css`
+    margin-bottom: 8px;
+`;
+const captionStyle = css`
+    padding: 8px 10px 0;
+    ${textSans(1)};
+    word-wrap: break-word;
+    color: ${palette.neutral[46]};
+`;
+
+export const YoutubeBlockComponent: React.FC<{
+    element: YoutubeBlockElement;
+    pillar: Pillar;
+}> = ({ element, pillar }) => {
+    // https://www.ampproject.org/docs/reference/components/amp-youtube
+    // https://developers.google.com/youtube/player_parameters
+    const attributes = {
+        id: `gu-video-youtube-${element.id}`,
+        'data-videoid': element.assetId,
+        layout: 'responsive',
+        width: '16',
+        height: '9',
+        'data-param-modestbranding': true, // Remove YouTube logo
+        'data-param-rel': '0', // Show the channel's related videos
+        'data-param-listType': 'playlist', // Related videos from playlist
+        'data-param-list': element.channelId, // Use specific channel ID for related videos
+    };
+
+    const iconStyle = css`
+        fill: ${pillarPalette[pillar].main};
+        padding-right: 3px;
+    `;
```

Comment:
Interesting. Are we inlining styles inside the component anywhere else? I think elsewhere we've used the [`pillarMap` approach](https://github.com/guardian/dotcom-rendering/blob/897b79abfe0e876be628b2dace5b92a430343436/packages/frontend/amp/components/TopMeta.tsx#L67). I wonder whether we should standardise on one approach... maybe we can discuss this in our Great Big Code Review next quarter. What do you think?

Reply:
I...do not have strong opinions on this to be honest! Is one is substantially quicker then let's do that, but I haven't tested.

Reply:
But agree that standardisation is beneficial.

Reply:
I'm thinking of it more from a developer rather than performance perspective:

- is it better to keep all styles together?
- should we keep styles close to or inside the component?
- is `pillarMap` discoverable / easy enough to understand? Will documentation help?

Reply:
I stole this from the image component (on my phone so can’t link sorry) - so there’s form, however I did wonder if there should be a caption component though we’ve not got any that go that small (do we repeat code or do we split out everything repeated or do we split on the third repetition?). I would say maybe we need linting on something to point at pillarMap as suggested use?